### PR TITLE
Plot features histograms #33

### PR DESCRIPTION
**Pull request by apoman38**

### CODE REVIEW

**Comment by tyiannak** [comment]
1. @apoman38 please resolve any conflicts with master before sending a pull request
2. always describe what the PR is about: what it does, how should one do the QA etc
3. make script names more meaningful: train.py should train a model and feature_selection should do feature selection. 
4. Add comments and docstrings to explain your functions.

**Comment by apoman38** [comment]
Sorry for my carelessness and for the wrong implementation of the task. I will be more specific about what was implemented. 

Initially, in the analyze_visual.py file in the dir_process_video() function, I added to save and return the names of the features. Then in train.py, it was added to feature_extraction() to store the names of the features in dictionary format.
A new file named feature_selection.py has been created which plots the feature histograms.

Please check if the feature_selection.py plots in the right form the histograms and let me know if changes need to be made.

File: analyze_visual/analyze_visual.py [hunk 1]
```diff
@@ -27,12 +21,9 @@
 import collections
 import sys
 import os.path
-sys.path.insert(0, os.path.join(os.path.dirname(
-    os.path.realpath(__file__)), "../"))
-
-from analyze_visual.object_detection import detection_utils as dutils
-from analyze_visual.object_detection import generic_model as gmodel
-from analyze_visual.utils import *
+from object_detection import detection_utils as dutils
```
Comment (tyiannak) [inline analyze_visual/analyze_visual.py:24]: please revert to the previous way these were imported

File: analyze_visual/feature_selection.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    #Save features and labels to list 
+    for key, value in x.items():
+        features.append(value)
+        class_names.append(key)
+
+    #Insert features names to numpy array
+    values = fname.values()
+    value_iterator = iter(values)
+    fnames = next(value_iterator)
+
```
Comment (tyiannak) [inline analyze_visual/feature_selection.py:97]: too many blank lines

File: analyze_visual/feature_selection.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            h = h.astype(float) / h.sum()
+            cbins = (bins[0:-1] + bins[1:]) / 2
+            scatter_1 = go.Scatter(x=cbins, y=h, name=class_names[fi],
+                                   marker=mark_prop, showlegend=(i == 0))
+            # (show the legend only on the first line)
+            figs.append_trace(scatter_1, int(i/n_columns)+1, i % n_columns+1)
+    for i in figs['layout']['annotations']:
+        i['font'] = dict(size=10, color='#224488')
+    plotly.offline.plot(figs, filename="report.html", auto_open=True)
+
```
Comment (tyiannak) [inline analyze_visual/feature_selection.py:65]: add blank line

**Comment by apoman38** [comment]
Please run the **feature_selection.py** with the command `python3 feature_selection.py -v dataset/Aerial dataset/Tilt` where dataset/Aerial, dataset/Tilt the directories of video files.
A file named **report.html** must be created after the run of feature_selection.py.
In this Html file, they are histograms of every feature separately.

**Pull request merged**